app/uk/gov/hmrc/customs/declaration/connectors/UpscanInitiateConnector.scala

```diff
@@ -35,24 +34,24 @@ class UpscanInitiateConnector @Inject()(http: HttpClient,
 
   private val configKey = "upscan-initiate"
 
-  def send[A](payload: UpscanInitiatePayload, apiVersion: ApiVersion)(implicit vupr: ValidatedUploadPayloadRequest[A]): Future[HttpResponse] = {
+  def send[A](payload: UpscanInitiatePayload, apiVersion: ApiVersion)(implicit vupr: ValidatedUploadPayloadRequest[A]): Future[UpscanInitiateResponsePayload] = {
     val config = Option(serviceConfigProvider.getConfig(s"${apiVersion.configPrefix}$configKey")).getOrElse(throw new IllegalArgumentException("config not found"))
     post(payload, config.url)
   }
 
   private def post[A](payload: UpscanInitiatePayload, url: String)(implicit vupr: ValidatedUploadPayloadRequest[A]) = {
 
     implicit val hc: HeaderCarrier = HeaderCarrier()
-    implicit val payloadFormat: OFormat[UpscanInitiatePayload] = Json.format[UpscanInitiatePayload]
+
     logger.debug(s"Sending request to upscan initiate service. Url: $url Payload: ${payload.toString}")
-    http.POST[UpscanInitiatePayload, HttpResponse](url, payload)
-      .recoverWith {
-        case httpError: HttpException => Future.failed(new RuntimeException(httpError))
-      }
-      .recoverWith {
-        case e: Throwable =>
-          logger.error(s"Call to upscan initiate failed. url=$url")
-          Future.failed(e)
-      }
+    val eventualResponse = http.POST[UpscanInitiatePayload, UpscanInitiateResponsePayload](url, payload)
+    eventualResponse.map(res =>
```

Comment:
Still needs changing. What is the benefit of calling map here? 
Also in the line above you can set the type to Future[UpscanInitiateResponsePayload] so its clear what we are working with.

Reply:
:+1: